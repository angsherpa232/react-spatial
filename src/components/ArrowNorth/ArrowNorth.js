import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import ArrowNorthSimple from '../../images/arrowNorth.svg';
import ArrowNorthCircle from '../../images/arrowNorthCircle.svg';

const propTypes = {
  /**
   * CSS class of the button.
   */
  className: PropTypes.string,

  /**
   * Rotation of the north arrow in degrees.
   */
  rotationOffset: PropTypes.number,

  /**
   * Display circle around the north arrow.
   */
  circled: PropTypes.bool,
};

const defaultProps = {
  className: 'tm-arrow-north',
  rotationOffset: 0,
  circled: false,
};

/**
 * This component displays an arrow pointing the North of the map.
 */
class ArrowNorth extends PureComponent {
  render() {
    const { className, circled, rotationOffset } = this.props;

    return (
      <div
        className={className}
        style={{ transform: `rotate(${rotationOffset}deg)` }}
      >
        {circled ? (
          <ArrowNorthCircle className={className} />
        ) : (
          <ArrowNorthSimple className={className} />
        )}
      </div>
    );
  }
}

ArrowNorth.propTypes = propTypes;
ArrowNorth.defaultProps = defaultProps;

export default ArrowNorth;
