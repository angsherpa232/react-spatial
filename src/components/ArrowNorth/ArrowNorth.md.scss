.tm-arrow-north-example {
  position: relative;

  .tm-map {
    width: 100%;
    height: 100%;
    flex-grow: 1;
  }

  .tm-arrow-north {
    height: 80px;
    width: 80px;
  }
}
