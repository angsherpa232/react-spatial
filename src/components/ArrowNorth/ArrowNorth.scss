.tm-arrow-north {
  bottom: 5px;
  right: 10px;
  height: 40px;
  width: 40px;
  position: absolute;
}
