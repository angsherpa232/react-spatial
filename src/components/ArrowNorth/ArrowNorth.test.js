import 'jest-canvas-mock';
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArrowNorth from './ArrowNorth';

configure({ adapter: new Adapter() });

describe('CanvasSaveButton', () => {
  test('should match snapshot.', () => {
    const component = renderer.create(<ArrowNorth className="test-class" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot rotated.', () => {
    const component = renderer.create(<ArrowNorth rotationOffset={45} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot with circle.', () => {
    const component = renderer.create(<ArrowNorth circled />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
