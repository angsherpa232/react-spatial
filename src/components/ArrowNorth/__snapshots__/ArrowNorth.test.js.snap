// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasSaveButton should match snapshot rotated. 1`] = `
<div
  className="tm-arrow-north"
  style={
    Object {
      "transform": "rotate(45deg)",
    }
  }
>
  <svg
    className="tm-arrow-north"
    data-file-name="SvgArrowNorth"
  />
</div>
`;

exports[`CanvasSaveButton should match snapshot with circle. 1`] = `
<div
  className="tm-arrow-north"
  style={
    Object {
      "transform": "rotate(0deg)",
    }
  }
>
  <svg
    className="tm-arrow-north"
    data-file-name="SvgArrowNorthCircle"
  />
</div>
`;

exports[`CanvasSaveButton should match snapshot. 1`] = `
<div
  className="test-class"
  style={
    Object {
      "transform": "rotate(0deg)",
    }
  }
>
  <svg
    className="test-class"
    data-file-name="SvgArrowNorth"
  />
</div>
`;
